### PR TITLE
Adding basic login/logout.

### PR DESCRIPTION
**Pull request by moniika**
Added basic styled login/logout button and logic to use the currently logged-in user's id.
Refactoring of common template code between basic and online.

### CODE REVIEW

File: appengine/js/lib-interface.js [hunk 1]
```diff
@@ -65,13 +66,63 @@ BlocklyInterface.init = function() {
...
+        if (responseData['login_url']) {
+          templateData.loginUrl = responseData['login_url'];
+        } else if (responseData['logout_url']) {
+          templateData.logoutUrl = responseData['logout_url'];
+        }
+        var loginButton =
+            soy.renderAsFragment(BlocklyGames.soy.loginButton, templateData,
+                null);
+        farSide.appendChild(loginButton);
+
```
Comment (alschmiedt) [inline appengine/js/lib-interface.js]: nit: remove extra line

File: appengine/template.soy [hunk 2]
```diff
@@ -74,6 +74,29 @@
   </div>
 {/template}
 
+
+
```
Comment (alschmiedt) [inline appengine/template.soy]: nit: remove extra lines.

File: appengine/js/lib-interface.js [hunk 3]
```diff
@@ -65,13 +66,63 @@ BlocklyInterface.init = function() {
     linkButton.style.display = 'none';
   }
 
+  var loginDiv = document.getElementById('login');
+  if (loginDiv) {
+    BlocklyInterface.setupLogin();
```
Comment (alschmiedt) [inline appengine/js/lib-interface.js]: If we have the loginDiv here could we pass this into setupLogin and use this to add the loginButton to?

File: appengine/js/lib-interface.js [hunk 4]
```diff
@@ -65,13 +66,63 @@ BlocklyInterface.init = function() {
...
+ * Fire a new AJAX request.
+ * @param {string} url URL to fetch.
+ * @param {Object.<string, string>} data Body of data to be sent in request.
+ * @param {string=} [method="POST"] The HTTP request method to use, such as
+ *    "GET", "POST", "PUT", "DELETE".
+ */
+BlocklyInterface.makeRequest = function(url, data, onLoadCallback, method) {
+  // TODO(kozbial) refactor duplicated ajax request code.
+  var xhr = new XMLHttpRequest();
+  var method = method || 'POST';
```
Comment (alschmiedt) [inline appengine/js/lib-interface.js]: just curious why this defaults to post?

File: appengine/js/lib-interface.js (hunk 3 above)
Comment (moniika) [reply appengine/js/lib-interface.js]: This is actually leftover from previous version where I used loginDiv. Current logic does not use the div and instead adds the button the html page to right side of navbar (id "farSide"). I will update code to remove the old login divs

File: appengine/js/lib-interface.js (hunk 1 above)
Comment (moniika) [reply appengine/js/lib-interface.js]: Done

File: appengine/js/lib-interface.js (hunk 4 above)
Comment (moniika) [reply appengine/js/lib-interface.js]: most/all currently use post.

File: appengine/template.soy (hunk 2 above)
Comment (moniika) [reply appengine/template.soy]: Done

File: appengine/js/lib-games.js [hunk 5]
```diff
@@ -183,6 +184,14 @@ BlocklyGames.isRtl = function() {
   return BlocklyGames.LANGUAGE_RTL.indexOf(BlocklyGames.LANG) != -1;
 };
 
+/**
+ * Is the current language (BlocklyGames.LANG) an RTL language?
```
Comment (alschmiedt) [inline appengine/js/lib-games.js]: update comment?

Comment (moniika) [reply appengine/js/lib-games.js]: Done.

**Pull request merged**